app/javascript/controllers/hide_devise_alert_controller.js

```diff
@@ -0,0 +1,18 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="hide-devise-alert"
+export default class extends Controller {
+  static targets = ["alert"];
+
+  connect() {
+    if (this.hasAlertTarget) {
+      setTimeout(() => {
+        this.hideAlert();
+      }, 3000);
+    }
+  }
+
+  hideAlert() {
+    this.alertTarget.style.display = "none";
+  }
+}
```

Comment:
You can remove the static targets and use `this.element` instead of `this.alertTarget` `this.hasAlertTarget`